### PR TITLE
Add upgrade instructions to v6 and above

### PR DESCRIPTION
**Pull request by ryyppy**
- [X] Add Upgrade guide in Docs
- [ ] ~~Add docs for record-as-objects~~
- [ ] ~~Refactor `Object` / `Object 2` docs to something more meaningful and communicate difference to records-as-objects feature~~

(Will do the rest in upcoming PRs)

### CODE REVIEW

**Comment by bobzhang** [comment]
ping me when it is good for review, thanks

**Comment by ryyppy** [comment]
@bobzhang Ready for review

File: docs/upgrade-to-v6-and-v7.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      support (introduced with `4.03`)
+- *BS 7.x and above*: based on OCaml 4.06.1 (stops 4.02.3 support)
+    - Adds the `Records-as-Objects` feature, which makes BS compile record type
+      values into JS objects instead of nested arrays (see blog [post 1](/blog/2019/11/18/whats-new-in-7)
+      and [post 2](/blog/2019/12/27/whats-new-in-7-cont))
+    - Ships with `refmt` version `3.5.1` with greatly improved syntax error messages (codename `reerror`)
+
+The biggest upside of upgrading to BS v6 and v7 is that we will finally be able
+to deprecate  underlying OCaml `4.02.3` code, which will make maintenance way
+easier (lots of code / processes can be removed and simplified). Also the OCaml
```
Comment (bobzhang) [inline docs/upgrade-to-v6-and-v7.md]: If we don't mention v6, this needs be rephrased?

File: docs/upgrade-to-v6-and-v7.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+- `decco`: [package.json for 4.06](https://github.com/reasonml-labs/decco/blob/master/ppx_src/package.json)
+
+### Step 2: Release artifacts and add a proper Note in the README
+
+Make sure to release the newly built binary on npm and make clear which PPX versions are compatible with BS6 / BS5 etc.
+For example, this is how `gentype` does it:
+
+```text
+bs-platform 7.0.0 or higher: use genType 3.2.0 or higher.
+bs-platform 6.2.0 or higher: use genType 3.0.0 or higher.
```
Comment (bobzhang) [inline docs/upgrade-to-v6-and-v7.md]: same here

File: docs/upgrade-to-v6-and-v7.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    "ocaml": "~4.6.1000",
+    "@opam/dune": "*",
+}
+```
+
+If you are using `migrate-parse-tree`, you might want to drop any AST upgrades
+from 4.02 -> 4.06 etc, making the native binary more lightweight. 
+
+**Tip:** If you still want to maintain a `4.02.3` based version (for legacy
+reasons), you can alternatively create a `esy.json` / `esy-406.json` file and
```
Comment (bobzhang) [inline docs/upgrade-to-v6-and-v7.md]: shall we mention we support conditional compilation per version somewhere

**Review by bobzhang — COMMENTED** [review]
In general lgmt modulo some minor comments, thanks

**Pull request merged**